controller/behavior/denormalizeable.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/**
+ * @package     LOGman
+ * @copyright   Copyright (C) 2011 - 2016 Timble CVBA. (http://www.timble.net)
+ * @license     GNU GPLv3 <http://www.gnu.org/licenses/gpl.html>
+ * @link        http://www.joomlatools.com
+ */
+
+/**
+ * Denormalizeable Controller Behavior
+ *
+ * Allows for de-normalization of activities entries
+ *
+ * @author  Arunas Mazeika <https://github.com/amazeika>
+ * @package Joomlatools\Component\LOGman
+ */
+class ComActivitiesControllerBehaviorDenormalizeable extends KControllerBehaviorAbstract
+{
+    /**
+     * The resources controller
+     *
+     * @var KModelInterface
+     */
+    protected $_controller;
+
+    public function __construct(KObjectConfig $config)
+    {
+        parent::__construct($config);
+
+        $this->_controller = $config->controller;
+    }
+
+    protected function _initialize(KObjectConfig $config)
+    {
+        $config->append(array(
+            'controller' => 'resource'
+        ));
+
+        if ($this->getIdentifier()->getPackage() != 'activities')
+        {
+            $aliases = array(
+                'com:activities.controller.resource' => array(
+                    'path' => array('controller'),
+                    'name' => 'resource'
+                )
+            );
+
+            foreach ($aliases as $identifier => $alias)
+            {
+                $alias = array_merge($this->getMixer()->getIdentifier()->toArray(), $alias);
+
+                $manager = $this->getObject('manager');
+
+                // Register the alias if a class for it cannot be found.
+                if (!$manager->getClass($alias, false)) {
+                    $manager->registerAlias($identifier, $alias);
+                }
+            }
+        }
+
+        parent::_initialize($config);
+    }
+
+    protected function _afterAdd(KControllerContextInterface $context)
+    {
+        $entity = $context->result;
+
+        if ($entity->action != 'delete')
+        {
+            $context = $this->getMixer()->getContext();
+
+            $context->entity = $entity;
+
+            $this->denormalize($context);
```

Comment:
Not a fan really, denormalising is not an action, you only change the format of the data, but the data is the same. This action will trigger also a lot of other things that don't need to get triggered. Can we go without an additional action for this?